static/js/muting_ui.js

```diff
@@ -16,6 +16,46 @@ exports.rerender = function () {
     }
 };
 
+exports.mute_notification = (function () {
+    var timeout,
+        animate = {
+            fadeOut: function ($mute) {
+                $mute.fadeOut(500).removeClass("show");
+            },
+            fadeIn: function ($mute) {
+                $mute.fadeIn(500).addClass("show");
+                clearTimeout(timeout);
+            }
+        },
+        event_added = false;
+
+    return function (stream, topic) {
+        var $mute_topic = $("#mute_topic_container"),
+            $exit = $("#mute_topic_container .exit-me");
+
+        if (!event_added) {
+            event_added = !event_added;
+
+            $exit.click(function () {
+                animate.fadeOut($mute_topic);
+            });
+
+            $mute_topic.find("#unmute").click(function () {
+                popovers.topic_ops.unmute(stream, topic);
+                animate.fadeOut($mute_topic);
+            });
+        }
+
+        $mute_topic.find(".topic").html(topic);
```

Comment:
I would consider moving this line of code before the callback section, so that structurally the code reads like this:

- populate the widget
- set up its handlers
- make it visible

I dunno, though, I can see the logic of setting the handlers first.